### ISSUE TITLE
When querying against a date, support integral unix times as well as Dates

### ISSUE THREAD

**Issue by dbkaplun** [balderdashy/waterline#1071]

**Comment by tjwebb** [comment]
You mean, just send in an integer timestamp like `1436247886000`? What's wrong with just using `new Date(1436247886000)`?

**Comment by dbkaplun** [comment]
I think `new Date()` is great, there's nothing wrong with it. I just think `{createdAt: {'<=': Date.now()-HOUR}}` is nicer than `{createdAt: {'<=': new Date(Date.now()-HOUR)}}`, don't you think?

**Comment by dmarcelino** [comment]
Dates are casted in [typecast.js#L207-L221](https://github.com/balderdashy/waterline/blob/master/lib/waterline/core/typecast.js#L207-L221). If @tjwebb  and @devinivy agree, I wouldn't oppose adding a rule such as:

```js
  else if (typeof value === 'number') {
    _value = new Date(value);
  }
```

I'm not a 100% this cast operation applies to both values being saved and to criteria values (as @dbkaplun shown in his example) though it should be.

**Comment by sailsbot** [comment]
Thanks for posting, @dbkaplun.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
